subsys/bluetooth/mesh/shell/shell_model_utils.c

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <bluetooth/mesh/models.h>
+#include "mesh/net.h"
+#include "mesh/access.h"
+
+struct bt_mesh_model *shell_model_first_get(uint16_t id)
+{
+	const struct bt_mesh_comp *comp = bt_mesh_comp_get();
+	struct bt_mesh_elem *elem;
+	struct bt_mesh_model *mod;
+
+	for (size_t i = 0; i < comp->elem_count; i++) {
+		elem = bt_mesh_elem_find(comp->elem[i].addr);
+		mod = bt_mesh_model_find(elem, id);
+		if (mod) {
+			return mod;
+		}
+	}
+
+	return NULL;
+}
+
+void shell_model_instances_get(uint16_t id, uint16_t *arr, uint8_t len)
+{
+	const struct bt_mesh_comp *comp = bt_mesh_comp_get();
+	struct bt_mesh_elem *elem;
+
+	for (size_t i = 0; i < len; i++) {
+		elem = bt_mesh_elem_find(comp->elem[i].addr);
+		if (bt_mesh_model_find(elem, id)) {
+			arr[i] = comp->elem[i].addr;
+		}
+	}
+}
+
+struct bt_mesh_model *shell_model_get(uint16_t id, uint16_t addr)
```

Comment:
It might be easier to work with this API if it uses element index, so we don't have to calculate the address